### PR TITLE
Add SenshiManga

### LINKED ISSUE #2466: Source to add Senshi manga
### Source name

Senshi manga

### Source link

https://senshimanga.com

### Source language

Spanish

### Other details

_No response_

### Acknowledgements

- [X] I have checked that the extension does not already exist by searching the [extension listing](https://keiyoushi.github.io/extensions/) and verified it does not appear there.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a meaningful title with the source name.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by bapeey**
Closes #2466

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

**Comment by AwkwardPeak7** [comment]
why make a multisrc with only one source?

**Comment by bapeey** [comment]
the API returns a different result depending on the site (e.g. https://lat-manga.com/), so I made multisrc just in case

File: src/es/senshimanga/src/eu/kanade/tachiyomi/extension/es/senshimanga/SenshiManga.kt [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+    override val supportsLatest = true
+
+    private val organizationDomain = baseUrl.substringAfter("://")
+
+    private val apiBaseUrl = "https://lat-manga.com"
+
+    private val json: Json by injectLazy()
+
+    override val client: OkHttpClient = network.client.newBuilder()
```
Comment (AwkwardPeak7) [inline src/es/senshimanga/src/eu/kanade/tachiyomi/extension/es/senshimanga/SenshiManga.kt]: use cloudflareClient

File: src/es/senshimanga/src/eu/kanade/tachiyomi/extension/es/senshimanga/SenshiManga.kt [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        .rateLimitHost(baseUrl.toHttpUrl(), 3)
+        .rateLimitHost(apiBaseUrl.toHttpUrl(), 3)
+        .build()
+
+    override fun headersBuilder(): Headers.Builder = Headers.Builder()
```
Comment (AwkwardPeak7) [inline src/es/senshimanga/src/eu/kanade/tachiyomi/extension/es/senshimanga/SenshiManga.kt]: use `super.headersBuilder`

File: src/es/senshimanga/src/eu/kanade/tachiyomi/extension/es/senshimanga/SenshiManga.kt [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+    override val name = "Senshi Manga"
+
+    override val baseUrl = "https://senshimanga.com"
+
+    override val lang = "es"
+
+    override val supportsLatest = true
+
+    private val organizationDomain = baseUrl.substringAfter("://")
```
Comment (AwkwardPeak7) [inline src/es/senshimanga/src/eu/kanade/tachiyomi/extension/es/senshimanga/SenshiManga.kt]: inline this

File: src/es/senshimanga/src/eu/kanade/tachiyomi/extension/es/senshimanga/SenshiMangaDto.kt [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+@Serializable
+class SeriesPayloadDto(
+    val data: SeriesListDataDto,
```
Comment (AwkwardPeak7) [inline src/es/senshimanga/src/eu/kanade/tachiyomi/extension/es/senshimanga/SenshiMangaDto.kt]: make a template class for data, `Data<T>(val data)`

File: src/es/senshimanga/src/eu/kanade/tachiyomi/extension/es/senshimanga/SenshiMangaDto.kt [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+class SeriesChapterDto(
+    private val title: String,
+    private val number: Float,
+    private val createdAt: String,
+) {
+    fun toSChapter(seriesSlug: String) = SChapter.create().apply {
+        name = "Capítulo ${number.toString().removeSuffix(".0")} - $title"
+        date_upload = try {
+            dateFormat.parse(createdAt)?.time ?: 0L
+        } catch (_: Exception) {
```
Comment (AwkwardPeak7) [inline src/es/senshimanga/src/eu/kanade/tachiyomi/extension/es/senshimanga/SenshiMangaDto.kt]: ParseException

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**